web/index.html

```diff
@@ -421,14 +527,49 @@ <h3>Used in</h3>
                     }
                     return n.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
                 },
-                loadChunk: function (chunk) {
-                    fetch(DATA_PREFIX + `chunk-${chunk}.json`)
-                        .then(response => { console.log("Loaded chunk", chunk); return response.json() })
+                loadChunk: async function (chunk) {
+                    return fetch(DATA_PREFIX + `chunk-${chunk}.json`)
+                        .then(response => {
+                            console.log("Loaded chunk", chunk);
+                            return response.json()
+                        })
                         .then(data => {
                             this.data.data = Object.freeze(Object.assign({}, this.data.data, data))
                             this.data.metadata.loadedChunks.push(chunk)
-                            console.log(this.data.data)
-                        })
+                        });
+                },
+                calcCraftingTree: async function (element) {
+                    if (this.data.costs[element] == 1) {
+                        return [];
+                    }
+
+                    let craftingList = [];
+                    let usedElements = new Set();
+                    let result = [];
+                    craftingList.push(element);
+
+                    while (craftingList.length > 0) {
+                        let thisElement = craftingList.shift();
+                        if (usedElements.has(thisElement)) {
+                            continue;
+                        }
+                        usedElements.add(thisElement);
+
+                        if (!this.data.data[thisElement] && thisElement != "") {
```

Comment:
I have done that in my tree branch update since most of this function needs to be rewritten for that anyways. 
For now since this is functionally identical, still readable, and will be overwritten soon, I think it's fine to have in for just this PR.

Reply:
Okay done